src/plugins/data/public/ui/query_editor/query_editor_extensions/query_editor_extension.tsx

```diff
@@ -145,10 +147,41 @@ export const QueryEditorExtension: React.FC<QueryEditorExtensionProps> = (props)
   }, []);
 
   useEffect(() => {
-    const subscription = props.config.isEnabled$(props.dependencies).subscribe((enabled) => {
-      if (isMounted.current) setIsEnabled(enabled);
-    });
-    return () => subscription.unsubscribe();
+    let subscription: Subscription;
+
+    const fetchEnabled = async () => {
+      const isEnabledObservable = await props.config.isEnabled$(props.dependencies);
+      subscription = isEnabledObservable.subscribe((enabled) => {
+        if (isMounted.current) setIsEnabled(enabled);
+      });
+    };
+
+    const setComponents = async () => {
+      const [
+        bannerResult,
+        componentResult,
+        queryControlButtonsResult,
+        bottomPanelResult,
+        actionBarButtonsResult,
+      ] = await Promise.all([
```

Comment:
if any component getter fails, this would cause all things to not load. should this be allSettled?

Reply:
good point. the other point to bring up is what to do if any fails, but since that was not handled prior, i will mark it as out of scope in my current pr. but based on our other comments, i think i can just undo this change